[DataPipe] Adding StreamWrapper to ensure file object will be closed

NivekT opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66559



cc @VitalyFedyunin @ejguan @NivekT

Differential Revision: [D31654026](https://our.internmc.facebook.com/intern/diff/D31654026)

NivekT committed:
[DataPipe] Adding StreamWrapper to ensure file object will be closed

[ghstack-poisoned]

NivekT committed:
Update on "[DataPipe] Adding StreamWrapper to ensure file object will be closed"




cc @VitalyFedyunin ejguan @NivekT

[ghstack-poisoned]

NivekT committed:
Update on "[DataPipe] Adding StreamWrapper to ensure file object will be closed"




cc @VitalyFedyunin ejguan @NivekT

[ghstack-poisoned]

ejguan reviewed (commented):
Could we first list all file streams that we plan to have? And ones that users potentially are going to use.
Then, we check how to align an API to `read` bytes or string from them.

ejguan reviewed (approved):
LGTM in general. Thanks!!

NivekT committed:
Update on "[DataPipe] Adding StreamWrapper to ensure file object will be closed"




cc @VitalyFedyunin ejguan @NivekT

Differential Revision: [D31654026](https://our.internmc.facebook.com/intern/diff/D31654026)

[ghstack-poisoned]

NivekT commented:
Due to the need to land to PyTorch and TorchData at the same time, closing this and land via [this PR instead](https://github.com/pytorch/pytorch/pull/66715).
